### DISCUSSION TITLE
unable to connect Trino superset after enabling ldap on trino

### DISCUSSION THREAD

**Discussion by Ankitchandre** [apache/superset#19086] [General]
I'm trying to connect my superset to trino. this worked fine, until we had to enable auth in trino and make it available behind a real domain trino.mycompany.com. my connection string looks like this:
trino://myuser:mypassword@trino.mycompany.com/catalog

Error on superset :

[SupersetError(message='(builtins.NoneType) None\n(Background on this error at: http://sqlalche.me/e/13/dbapi)', error_type=<SupersetErrorType.GENERIC_DB_ENGINE_ERROR: 'GENERIC_DB_ENGINE_ERROR'>, level=, extra={'engine_name': 'Trino', 'issue_codes': [{'code': 1002, 'message': 'Issue 1002 - The database returned an unexpected error.'}]})]

**Comment by srinify** [comment]
Hey Ankit! Github Discussions or the Superset Slack is the best place to get help with debugging. This doesn't seem to be a bug :]

**Reply by CarlosMgno** [reply]
I have this error when trying to connect with druid

**Reply by CarlosMgno** [reply]
![image](https://user-images.githubusercontent.com/99760368/173379569-fe1c4195-fd6a-4426-af29-ba04a72032d1.png)

**Reply by rohit-mobstac** [reply]
Having the same issue while connecting Apache Druid

**Comment by sbbagal13** [comment]
Any update on this , I am also facing same issue with Superset 2.0 and Trino 393

**Comment by RavazziniAndrea** [comment]
Same error with superset 2.0 and IBM DB2

**Comment by tarekabouzeid** [comment]
Hi,
We have Trino protected via LDAP and connecting to it via superset. in the connection advanced in the extra security we added the below
`
{
    "auth_method": "basic",
    "auth_params": {
    "username": "xxxxx",
    "password": "xxxx"
    }
}
`
Also as its multi user setup, we needed to mark "Impersonate logged in user (Presto, Trino, Drill, Hive, and GSheets)" True. Please keep in mind that the provided user must be able to impersonate the logged in superset user, so queries are actually ran with superset logged in user.

Thanks
